libs/damap/src/lib/components/dmp/dmp-actions/dmp-actions.component.spec.ts

```diff
@@ -99,17 +102,48 @@ describe('DmpActionsComponent', () => {
     expect(dialogs.length).toBe(0);
   });
 
-  it('should dispatch export dmp action', async () => {
-    spyOn(store, 'dispatch');
-    component.exportDmp();
+  it('should call dispatchExportDmp if funderSupported is true', async () => {
+    spyOn(component, 'dispatchExportDmp').and.callThrough();
+    spyOn(component, 'exportDmpTemplate').and.callThrough();
+
+    spyOn(component.dmpForm.controls.project, 'getRawValue').and.returnValue({
+      funderSupported: true,
+    });
+
+    component.exportDmpTemplate();
+    spyOn(store, 'dispatch').and.callThrough();
+    component.dispatchExportDmp();
 
     let dialogs = await loader.getAllHarnesses(MatDialogHarness);
     expect(dialogs.length).toBe(1);
-    await dialogs[0].close();
 
+    await dialogs[0].close();
     dialogs = await loader.getAllHarnesses(MatDialogHarness);
     expect(dialogs.length).toBe(0);
+    expect(component.exportDmpTemplate).toHaveBeenCalledTimes(1);
+    expect(component.dispatchExportDmp).toHaveBeenCalledTimes(1);
+  });
 
-    expect(store.dispatch).toHaveBeenCalledTimes(1);
+  it('should call dispatchExportDmp if funderSupported is false', async () => {
+    spyOn(component, 'dispatchExportDmpTemplate').and.callThrough();
+    spyOn(component, 'exportDmpTemplate').and.callThrough();
+
+    spyOn(component.dmpForm.controls.project, 'getRawValue').and.returnValue({
+      funderSupported: false,
+    });
+
+    component.exportDmpTemplate();
+    spyOn(store, 'dispatch').and.callThrough();
+    component.dispatchExportDmpTemplate(ETemplateType.FWF);
+
+    let dialogs = await loader.getAllHarnesses(MatDialogHarness);
+    expect(dialogs.length).toBe(1);
+
+    await dialogs[0].close();
+    dialogs = await loader.getAllHarnesses(MatDialogHarness);
+
+    expect(dialogs.length).toBe(0);
+    expect(component.exportDmpTemplate).toHaveBeenCalledTimes(1);
+    expect(component.dispatchExportDmpTemplate).toHaveBeenCalledTimes(1);
   });
```

Comment:
These tests might not cover the logic, which has to be tested. Whether `dispatchExportDmp` or `dispatchExportDmpTemplate` is called, depends on the project. The respective function should also be called automatically before the dialog is closed and not manually (as in the testcase).